src/software/ai/hl/stp/tactic/crease_defender/crease_defender_fsm.cpp

```diff
@@ -154,3 +190,41 @@ std::optional<Point> CreaseDefenderFSM::findDefenseAreaIntersection(
 
     return std::nullopt;
 }
+
+bool CreaseDefenderFSM::ballNearbyWithoutThreat(const Update& event)
+{
+    Point robot_position   = event.common.robot.position();
+    double ball_position_x = event.common.world_ptr->ball().position().x();
+    std::optional<Robot> nearest_enemy =
+        event.common.world_ptr->enemyTeam().getNearestRobot(robot_position);
+    if (nearest_enemy)
+    {
+        // Get the ball if ball is closer to robot than enemy threat by threshold ratio
+        double ball_distance =
+            distance(robot_position, event.common.world_ptr->ball().position());
+        double nearest_enemy_distance =
+            distance(robot_position, nearest_enemy->position());
+
+        return ball_distance < nearest_enemy_distance * MAX_GET_BALL_RATIO_THRESHOLD &&
+               ball_position_x < 0 && ball_distance <= MAX_GET_BALL_RADIUS_M &&
+               event.common.world_ptr->ball().velocity().length() <=
+                   MAX_BALL_SPEED_TO_GET_MS;
```

Comment:
There's a lot going on in this return value. Consider flipping each of the comparisons and checking them one by one in an if statement (returning false if the condition doesn't hold), or having the result of each comparison stored in a bool variable and the checking that all of them are true.